tracks/ruby/exercises/matrix/README.md

```diff
@@ -0,0 +1,35 @@
+# Matrix
+
+## Reasonable Solutions
+
+```ruby
+class Matrix
+  attr_reader :rows, :columns
+  def initialize(input)
+    @rows = split_input(input)
+    @columns = rows.transpose
+  end
+
+  private
+
+  def split_input(string)
+    string.lines.map { |line| split_line(line) }
+  end
+
+  def split_line(line)
+    line.split.map(&:to_i)
+  end
+end
+```
+
+Many solutions will not split out the input parsing into methods like this. We don't yet have consensus among mentors if this is important or not. If the methods are split, the above example does demonstrate how the methods could be flagged as private. Some Ruby projects insist on marking as private methods that are not part of the public, tested interface.
+
+## Common Suggestions
+
+There's no need to overload a student with all of these suggestions, so here is a list ordered roughly by most important suggestions first:
+
+* For students who wrote code that interleaves string manipulation with data persistance, encourage them to do those steps sequentially
+* Suggest the student look at other methods on `String` that could simplify their code if they don't use `String#lines`
+* More generally, for any solutions that has matching based on a regular expression or explicitly decleared string or substring, encourage them to find a solution that does not require either
```

Comment:
I'd vote to make this the most important suggestion. (In context: In Acronym, students found better alternatives to `split` for special cases. The Matrix exercise is the perfect example of what `split` is doing better than one's own solution.)

Reply:
I agree with your suggestion to move it up in the list.

I have not seen many students who interleave string manipulation with accumulating data into the variables. Ideally, students wouldn't still be doing that by the time they get to this exercise. I definitely have seen it, though, and so I feel it's worth putting first. Do you think these are good enough reasons to move this third suggestion upward but not up to the top?

Reply:
It looks like 'interleaving' would imply a too complex solution in the first place, therefore I'd vote to put this one first. Don't you think so?
The `interleaving` is something I'd rather address _after_ the code is refactored.